source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+|Symbol              |{"$symbol": *string*}                                     |
++--------------------+----------------------------------------------------------+
+|String              |*string*                                                  |
++--------------------+----------------------------------------------------------+
+|Int32               |{"$numberInt": <32-bit signed integer as a *string*>      |
++--------------------+----------------------------------------------------------+
+|Int64               |{"$numberLong": <64-bit signed integer as a *string*>     |
++--------------------+----------------------------------------------------------+
+|Double              |{"$numberDouble": <64-bit binary floating point as a      |
+|                    |*string*>                                                 |
++--------------------+----------------------------------------------------------+
+|Decimal128          |{"$numberDecimal": <decimal as a *string*>} [#]_          |
++--------------------+----------------------------------------------------------+
+|Binary              |{"$binary": <base64-encoded (with padding as ``=``)       |
+|                    |payload as a *string*, "$type": <BSON binary type as a    |
+|                    |one- or two-character *hex string*}                       |
++--------------------+----------------------------------------------------------+
+|Code                |{"$code": *string*}                                       |
++--------------------+----------------------------------------------------------+
+|CodeWScope          |{"$code": *string*, "$scope": *Document*}                 |
++--------------------+----------------------------------------------------------+
+|Document            |*object* (with Extended JSON extensions)                  |
++--------------------+----------------------------------------------------------+
+|Timestamp           |{"$timestamp": <64-bit unsigned integer as a *string*>}   |
++--------------------+----------------------------------------------------------+
+|Regex               |{"$regex": *string*, "$options": <BSON regex options as a |
+|                    |*string* or "" [#]_}                                      |
++--------------------+----------------------------------------------------------+
+|DBPointer           |{"$dbPointer": {"$ref": <namespace [#]_ as a *string*>,   |
+|                    |"$id": *ObjectId*}                                        |
++--------------------+----------------------------------------------------------+
+|Datetime            |{"$date": {"$numberLong": <64-bit signed integer giving ms|
+|                    |since the epoch, as a *string*>}                          |
++--------------------+----------------------------------------------------------+
+|DBRef [#]_          |{"$ref": <collection name as a *string*>, "$id":          |
+|                    |<extended JSON for the id>}                               |
```

Comment:
DBPointer is limited to referring to an ObjectID as the 12-byte sequence is part of its structure; however, DBRef is flexible and can specify any type in its `$id` property.